tool/site/client/src/components/pagination.tsx

```diff
@@ -0,0 +1,87 @@
+import React from "react"
+import classnames from "classnames"
+import {usePagination, DOTS} from "./usePagination"
+
+interface PaginationProps {
+  totalCount: number
+  currentPage: number
+  pageSize: number
+  onPageChange: (pageChangeNumber: number) => void,
+  siblingCount?: number
+}
+
+const Pagination = (props: PaginationProps): any => {
+  const {
+    onPageChange,
+    totalCount,
+    siblingCount = 8,
+    currentPage,
+    pageSize
+  } = props
+
+  const paginationRange = usePagination({
+    currentPage,
+    totalCount,
+    siblingCount,
+    pageSize
+  })
+
+  // If there are less than 2 times in pagination range we shall not render the component
+  if (currentPage === 0 || paginationRange.length < 2) {
+    return null
+  }
+
+  const onNext = () => {
+    onPageChange(currentPage + 1)
+  }
+
+  const onPrevious = () => {
+    onPageChange(currentPage - 1)
+  }
+
+  const lastPage = paginationRange[paginationRange.length - 1]
+
+  return (
+    <ul className="pagination-container">
+      {/* Left navigation arrow */}
+      <li
+        className={classnames("pagination-item", {
+          disabled: currentPage === 1
+        })}
+        onClick={onPrevious}
+      >
+        <div className="arrow left" />
+      </li>
+      {paginationRange.map(pageNumber => {
+
+        // If the pageItem is a DOT, render the DOTS unicode character
+        if (pageNumber === DOTS) {
+          return <li className="pagination-item dots">&#8230;</li>
+        }
+
+        // Render our Page Pills
+        return (
+          <li
+            className={classnames("pagination-item", {
+              selected: pageNumber === currentPage
+            })}
+            onClick={() => onPageChange(pageNumber as number)}
```

Comment:
I think we can get rid of this `as` if we change the type in `usePagination`, I'll leave a comment there